climada/util/api_client.py

```diff
@@ -254,12 +256,24 @@ class NoResult(Exception):
     class NoConnection(Exception):
         """To be raised if there is no internet connection and no cached result."""
 
+    @staticmethod
+    def _host_and_port_from_url(url):
+        urlpat = r"http(s?)://([^:/]+)(:\d+)?(/|$)"
+        match = re.match(urlpat, url)
+        if match:
+            host = match.group(2)
+            port = int(match.group(3)[1:]) if match.group(3) else 443 if match.group(1) else 80
+        else:
+            raise ValueError(f'URL not as expected, cannot figure out host and port from "{url}"')
+        return host, port
```

Comment:
No need to use a regex, there is a function to extract url components: [urllib.parse.urlsplit](https://docs.python.org/3/library/urllib.parse.html#urllib.parse.urlsplit)
```suggestion
        url_split = urlsplit(url)
        port = url_split.port
        if port is None:
            port = 443 if url_split.scheme == "https" else 80
        return url_split.netloc, port
```

Reply:
cool 😎